fix: "Copy Image" menu item correctly copies images

sallto opened the pull request:
fixes #8348

The fundamental problem was the asset protocol, since this is not known by outside applications. 

I tested it with a few different programs and it seems to work every time

situ2001 commented on src/main/frontend/components/block.cljs:
```diff
@@ -301,11 +302,10 @@
         (merge
          {:loading "lazy"
-          :src     src
+          :src     image-src
```
Thanks for the fix; it works fine for me on the desktop version, too. should figure out why the assets protocol href was left here.

Some relevant issues and PRs:

- https://github.com/logseq/logseq/issues/7021
- https://github.com/logseq/logseq/pull/6475

sprocketc reviewed (commented):
Confirmed that it fixes the issue that can also be reproduced within Logseq. Before the fix, using the native context menu to copy an image and then paste it in the app, produced an invalid image source. The assets protocol was included which resulted in a duplicated path on render.

One potential remaining issue is that the location of the pasted image will be an absolute path. Also, the path now contains the file protocol, and clicking on the filename to open the file directory won't work.

sallto commented:
Ok I think I found a better solution now. Instead of messing with the imgs src-url, I modified the copy method and replaced the file:// protocol there

sprocketc reviewed (approved):
Thanks for resolving the remaining issues :heart:  
I tested this and it looks like it's working properly.

Note for the rest of the team:
Copy/pasting an image will now always create a new asset from the initial image data. This is consistent with the behavior of the copy button on the image overlay actions, which is not exactly the same with what `copyImageAt` was doing.

logseq-cldwalker reviewed (approved):
@sallto Thanks for the fix! 🐛 🔥 Confirmed this fixes copying with a gif and notes app on osx as well as within the app as @sprocketc noted. Interestingly this wasn't an issue when I QAed on a google drive page
